src/table.jl

```diff
@@ -289,6 +289,16 @@ Base.getindex(t::NextTable, i::Integer) = getindex(t.columns, i)
 Base.getindex(t::NextTable, i::Colon) = copy(t)
 Base.endof(t::NextTable) = length(t)
 
+function Base.view(t::NextTable, I)
+    issorted(I) || throw(ArgumentError("`view` called with unsorted index."))
+    NextTable(
+        view(t.columns, I),
+        t.pkey,
+        Perm[],
+        fill(Nullable{Float64}(), length(t.columns)),
```

Comment:
I think it's important to keep the primary keys, but I'm not sure what `Perm` does and whether it can be kept.

Reply:
You can throw away `Perm`s as you wish. It's not exposed functionality right now anyway.

Reply:
`Perm` stores a permutation (output of `sortpermby`) and which columns this sortperm was performed by. So if you're repeatedly grouping or joining by the same subset of columns you can use `join(x,y, cache=true)` and have that be cached...